client/analytics/report/products/table-variations.js

```diff
@@ -0,0 +1,198 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { map, orderBy } from 'lodash';
+
+/**
+ * WooCommerce dependencies
+ */
+import { Link, TableCard } from '@woocommerce/components';
+import { formatCurrency, getCurrencyFormatDecimal } from '@woocommerce/currency';
+import { appendTimestamp, getCurrentDates } from '@woocommerce/date';
+import { getNewPath, getTimeRelatedQuery, onQueryChange } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import ReportError from 'analytics/components/report-error';
+import { getFilterQuery, getReportChartData } from 'store/reports/utils';
+import { QUERY_DEFAULTS } from 'store/constants';
+
+class VariationsReportTable extends Component {
+	getVariationName( variation ) {
+		return variation.attributes.reduce( ( desc, attribute, index, arr ) => {
+			desc += `${ attribute.option }${ arr.length === index + 1 ? '' : ', ' }`;
+			return desc;
+		}, variation.product_name + ' / ' );
+	}
+
+	getHeadersContent() {
+		return [
+			{
+				label: __( 'Product / Variation Title', 'wc-admin' ),
+				key: 'name',
+				required: true,
+				isLeftAligned: true,
+			},
+			{
+				label: __( 'Items Sold', 'wc-admin' ),
+				key: 'items_sold',
+				required: true,
+				defaultSort: true,
+				isSortable: true,
+				isNumeric: true,
+			},
+			{
+				label: __( 'G. Revenue', 'wc-admin' ),
+				key: 'gross_revenue',
+				required: true,
+				isSortable: true,
+				isNumeric: true,
+			},
+			{
+				label: __( 'Orders', 'wc-admin' ),
+				key: 'orders_count',
+				isSortable: true,
+				isNumeric: true,
+			},
+			{
+				label: __( 'Status', 'wc-admin' ),
+				key: 'stock_status',
+			},
+			{
+				label: __( 'Stock', 'wc-admin' ),
+				key: 'stock',
+			},
+		];
+	}
+
+	getRowsContent( data = [] ) {
+		const { stockStatuses } = wcSettings;
+		const { query } = this.props;
+		const timeRelatedQuery = getTimeRelatedQuery( query );
+
+		return map( data, row => {
+			const {
+				items_sold,
+				gross_revenue,
+				orders_count,
+				stock_status = 'outofstock',
+				stock_quantity = '0',
```

Comment:
Should this be a `0` / number?

Reply:
Most likely, yes. This value will eventually be returned by the API. Other table queries use a string, so I'll keep it this way for consistency